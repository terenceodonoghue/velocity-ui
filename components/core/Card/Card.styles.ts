import { css, FunctionInterpolation, Theme } from '@emotion/react';
import { em, rgba } from 'polished';

export const card: FunctionInterpolation<Theme> = ({ palette }) => css`
  display: block;
  border-radius: 1px;
  border: solid 1px ${rgba(palette.accent, 0.08)};
  box-shadow: 0 10px 20px 0 ${rgba(palette.accent, 0.07)};
  margin: 12px;
  padding: 32px 24px;
  background-color: ${palette.neutral[50]};
`;

export const heading: FunctionInterpolation<Theme> = ({
  palette,
  typography,
}) => `
  margin: 0 0 26px 0;
  color: ${palette.neutral[600]};
  font-size: ${em(13, typography.fontSize)};
  font-weight: ${typography.fontWeight.regular};
  letter-spacing: 1.2px;
  text-transform: uppercase;
`;
