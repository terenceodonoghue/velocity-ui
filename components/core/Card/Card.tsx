import React, { forwardRef } from 'react';
import * as css from './Card.styles';
import type { CardProps } from './Card.types';

const Card = forwardRef<HTMLDivElement, CardProps>(
  ({ children, heading, ...props }, ref) => (
    <div css={[css.card]} data-testid="card" ref={ref} {...props}>
      {heading && (
        <h3 css={css.heading} data-testid="card-heading">
          {heading}
        </h3>
      )}
      {children}
    </div>
  ),
);

export default Card;
