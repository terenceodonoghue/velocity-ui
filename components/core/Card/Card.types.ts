import { HTMLAttributes } from 'react';

export interface CardProps extends HTMLAttributes<HTMLDivElement> {
  heading?: string;
}
