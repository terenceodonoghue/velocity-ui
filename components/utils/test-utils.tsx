import React, { FunctionComponent, ReactElement } from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';
import ThemeProvider from '~/components/core/ThemeProvider';

const Providers: FunctionComponent = ({ children }) => {
  return <ThemeProvider>{children}</ThemeProvider>;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
): RenderResult => render(ui, { wrapper: Providers, ...options });

export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';

export { customRender as render };
