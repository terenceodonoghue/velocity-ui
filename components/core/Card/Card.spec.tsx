import React from 'react';
import faker from 'faker';
import { render } from 'test-utils';
import Card from './Card';

describe('Card', () => {
  it('renders content', () => {
    const content = faker.hacker.phrase();
    const { container, getByTestId, queryByTestId } = render(
      <Card>{content}</Card>,
    );

    expect(getByTestId('card')).toHaveTextContent(content);
    expect(queryByTestId('card-heading')).toBeNull();
    expect(container).toMatchSnapshot();
  });

  it('renders content with heading', () => {
    const content = faker.hacker.phrase();
    const heading = faker.hacker.phrase();

    const { container, getByTestId } = render(
      <Card heading={heading}>{content}</Card>,
    );

    expect(getByTestId('card')).toHaveTextContent(content);
    expect(getByTestId('card-heading')).toHaveTextContent(heading);
    expect(container).toMatchSnapshot();
  });
});
