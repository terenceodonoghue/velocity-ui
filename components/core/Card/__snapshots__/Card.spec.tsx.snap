// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders content 1`] = `
.emotion-0 {
  display: block;
  border-radius: 1px;
  border: solid 1px rgba(46,91,255,0.08);
  box-shadow: 0 10px 20px 0 rgba(46,91,255,0.07);
  margin: 12px;
  padding: 32px 24px;
  background-color: #fff;
}

<div>
  <div
    class="emotion-0"
    data-testid="card"
  >
    The SMS program is down, synthesize the redundant feed so we can synthesize the SQL hard drive!
  </div>
</div>
`;

exports[`Card renders content with heading 1`] = `
.emotion-0 {
  display: block;
  border-radius: 1px;
  border: solid 1px rgba(46,91,255,0.08);
  box-shadow: 0 10px 20px 0 rgba(46,91,255,0.07);
  margin: 12px;
  padding: 32px 24px;
  background-color: #fff;
}

.emotion-1 {
  margin: 0 0 26px 0;
  color: #8798ad;
  font-size: 0.8666666666666667em;
  font-weight: 400;
  letter-spacing: 1.2px;
  text-transform: uppercase;
}

<div>
  <div
    class="emotion-0"
    data-testid="card"
  >
    <h3
      class="emotion-1"
      data-testid="card-heading"
    >
      You can't input the array without navigating the redundant GB alarm!
    </h3>
    You can't quantify the interface without parsing the neural ADP feed!
  </div>
</div>
`;
